src/SpinnerKernel.java

```diff
@@ -0,0 +1,168 @@
+
+    //Also, @Override makes it so the compiler picks the correct method, right?
+
+    //build this on top of Map
+    //Is this an ok way to approach building this component, or is there a
+    //better way?
+
+    //Should I just do the NoteCard component instead?
+
+    //I think this would be similar to a linked array, where one holds the
+    //values of the user and the other holds a counter of it.
+
+    //I don't know how effecient this will be oops.
+
+    //these should not be within kernel but in secondary I think.
+    //I need to study up my chart of extends and implements.
+
+    /*
+     * Kernel methods ---------------------------------------------------------
+     */
+
+    //instance that the slice value has not been seen yet, given a value
+    @Override
+    public final void addSlice(K key, V value) {
+        assert key != null : "Violation of: key is not null";
+        assert value != null : "Violation of: value is not null";
+        assert !this.hasKey(key) : "Violation of: key is not in DOMAIN(this)";
+
+        this.Spinner.add(key, value);
+    }
+
+    //instance that the slice value has not seen, given a value
+    @Override
+    public final void addSlice(K key, V value) {
+        assert key != null : "Violation of: key is not null";
+        assert value != null : "Violation of: value is not null";
+        assert !this.hasKey(key) : "Violation of: key is not in DOMAIN(this)";
+
+        Map.Pair<K, V> increment = this.Spinner.remove(key);
+        V newValue = increment.value() + value;
+        this.Spinner.add(increment.key(), newValue);
+    }
+
+    //instance that key has not been seen yet, not given a value
+    @Override
+    public final void addSlice(K key) {
+        assert key != null : "Violation of: key is not null";
+        assert !this.hasKey(key) : "Violation of: key is not in DOMAIN(this)";
+
+        this.Spinner.add(key, 1);
+    }
+
+    //instance that key has been seen, not given a value
+    @Override
+    public final void addSlice(K key) {
+        assert key != null : "Violation of: key is not null";
+        assert this.hasKey(key) : "Violation of: key is in DOMAIN(this)";
+
+        Map.Pair<K, V> increment = this.Spinner.remove(key);
+        V newValue = increment.value() + 1;
+        this.Spinner(increment.key(), newValue);
+    }
+
+    //reduce slice by one
+    @Override
+    public final Pair<K, V> reduceSlice(K key) {
+        assert key != null : "Violation of: key is not null";
+        assert this.hasKey(key) : "Violation of: key is in DOMAIN(this)";
+
+        Pair<K, V> returnPair = this.Spinner.remove(key);
+        Value newValue = returnPair.value() - 1;
+        if(!newValue.equals(0)) {
+            //does not removes completely
+            this.Spinner.add(returnPair.key(), newValue);
+        }
+        return (returnPair.key(), newValue);
+    }
+
+    //reduce slice by a given value
+    @Override
+    public final Pair<K, V> reduceSlice(K key, V value) {
```

Comment:
I don't fully understand the point of this component, but if it's meant to be something like a roulette board or a wheel of fortune type design, then I wouldn't return the pair to the user. Chances are they just want the key or value. That way, the user never needs to know there is a map under the hood.